.github/workflows/ci.yml

```diff
@@ -84,3 +84,22 @@ jobs:
       run: cargo install cargo-audit
     - name: Run audit command
       run: cargo audit ${{ env.CARGO_AUDIT_BASE_FLAGS }}
+  credentials-scan:
+    name: Credentials scan
+    runs-on: ubuntu-latest
+    steps:
+    - name: PR commits + 1
+      # Adding +1 to fetch a base commit
+      run: echo "fetch_depth=$(( ${{ github.event.pull_request.commits }} + 1 ))" >> $GITHUB_ENV
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        # Checkout pull request HEAD commit instead of merge commit
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Ref: https://github.com/actions/checkout?tab=readme-ov-file#checkout-pull-request-head-commit-instead-of-merge-commit